app/controllers/shipit/api/ccmenu_controller.rb

```diff
@@ -0,0 +1,42 @@
+module Shipit
+  module Api
+    class CcmenuController < BaseController
+      require_permission :read, :stack
+
+      class NoDeploy
+        def id
+          0
+        end
+
+        def ended_at
+          Time.now.utc
+        end
+
+        def status
+          false
+        end
+
+        alias_method :running?, :status
+        alias_method :failed?, :status
+        alias_method :error?, :status
+        alias_method :success?, :status
+      end
+
+      def show
+        latest_deploy = stack.deploys.first || NoDeploy.new
```

Comment:
`stack.deploys` default ordering is `id ASC`, so if you want the most recent deploy, you need to use `stack.deploys.last`.

Also you likely want rollbacks as well, so `deploys_and_rollbacks` would be a better association: https://github.com/Shopify/shipit-engine/blob/e4162e2f0cd263723305e77eb5ccb76cbde09c5d/app/models/shipit/stack.rb#L31